source/structured-streaming.txt

```diff
@@ -37,6 +37,59 @@ Configuring a Write Stream to MongoDB
 
    tabs:
 
+     - id: java-sync
+       content: |
+
+         Specify write stream configuration settings on your streaming 
+         Dataset or DataFrame using the ``writeStream`` property. You 
+         must specify the following configuration settings to write 
+         to MongoDB:
+         
+         .. list-table::
+            :header-rows: 1
+            :stub-columns: 1
+            :widths: 10 40
+         
+            * - Setting
+              - Description
+         
+            * - ``writeStream.format()``
+              - The format to use for write stream data. Use 
+                ``mongodb``.
+         
+            * - ``writeStream.option()``
+              - Use the ``option`` method to specify your MongoDB 
+                deployment connection string with the 
+                ``spark.mongodb.connection.uri`` option key.
+         
+                You must specify a database and collection, either as 
+                part of your connection string or with additional 
+                ``option`` methods using the following keys:
+         
+                - ``spark.mongodb.database``
+                - ``spark.mongodb.collection``
+         
+            * - ``writeStream.outputMode()``
+              - The output mode to use. To view a list of all supported 
+                output modes, see `the Java outputMode documentation <https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/streaming/DataStreamWriter.html#outputMode-java.lang.String->`__.
+         
+         The following code snippet shows how to use the preceding 
+         configuration settings to stream data to MongoDB:
+
+         .. code-block:: java
+            :copyable: true
+            :emphasize-lines: 2-3, 6
+         
+            <streaming Dataset/ DataFrame>.writeStream()
+              .format("mongodb")
+              .option("spark.mongodb.connection.uri", <mongodb-connection-string>)
+              .option("spark.mongodb.database", <database-name>)
+              .option("spark.mongodb.collection", <collection-name>)
+              .outputMode("append");
+
+         For a complete list of methods, see the 
+         `Java Structured Streaming reference <https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/streaming/>`__.       
```

Comment:
Issue:
I wasn't sure where to go from the linked page as it was a generic directory listing of HTML docs.

Suggestion:
Is this the correct page? It seems to correspond with the methods you listed: https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/streaming/DataStreamWriter.html

Applies to all instances.

Reply:
I'll change these for all languages. I was trying to match the level of the python and scala links ([python example](https://spark.apache.org/docs/latest/api/python/reference/pyspark.ss.html)) and link out to a full reference, but it does make more sense (especially for Java) to link to the DataStreamReader/ Writer in the read/write sections.